model_compression_toolkit/core/common/quantization/set_node_quantization_config.py

```diff
@@ -194,26 +204,47 @@ def _create_node_candidates_qc(qc: QuantizationConfig,
                                node_qc_options_list: List[OpQuantizationConfig],
                                base_config: OpQuantizationConfig,
                                node: BaseNode,
-                               mixed_precision_enable: bool = False) -> List[CandidateNodeQuantizationConfig]:
+                               mixed_precision_enable: bool = False,
+                               nodes_to_manipulate_bit_widths: Dict = None) -> List[CandidateNodeQuantizationConfig]:
     """
     Create a list of candidates of weights and activation quantization configurations for a node.
 
     Args:
-        qc: Quantization configuration the quantization process should follow.
-        fw_info: Framework information (e.g., which layers should have their kernels' quantized).
-        weight_channel_axis: (Output, Input) channel index of the node's kernel.
-        node_qc_options_list: List of quantization configs of node.
-        base_config: Base quantization config for node.
-        node: A node to set quantization configuration candidates to.
-        mixed_precision_enable: is mixed precision enabled
+        qc (QuantizationConfig): Quantization configuration the quantization process should follow.
+        fw_info (FrameworkInfo): Framework information (e.g., which layers should have their kernels quantized).
+        weight_channel_axis (Tuple[int, int]): (Output, Input) channel index of the node's kernel.
+        node_qc_options_list (List[OpQuantizationConfig]): List of quantization configs of node.
+        base_config (OpQuantizationConfig): Base quantization config for node.
+        node (BaseNode): A node to set quantization configuration candidates to.
+        mixed_precision_enable (bool): Whether mixed precision is enabled. Defaults to False.
+        nodes_to_manipulate_bit_widths (Dict): Dictionary of nodes to manipulate bit-widths. Defaults to None.
 
     Returns:
-        List of candidates of weights quantization configurations to set for a node.
+        List[CandidateNodeQuantizationConfig]: List of candidates of weights quantization configurations to set for a node.
     """
 
     candidates = []
     node_attrs_list = node.get_node_weights_attributes()
 
+    if node in nodes_to_manipulate_bit_widths.keys():
```

Comment:
Please extract this part to a separate method "manually_overide_node_bitwidth" or something like that) and add lots of documentation explaining the purpose and the functionality